lib/wsdl.js

```diff
@@ -1206,7 +1206,98 @@ WSDL.prototype.toXML = function() {
   return this.xml || '';
 };
 
-WSDL.prototype.xmlToObject = function(xml) {
+WSDL.prototype.findInNamespace = function (splitPrefix_Name) {
+    if (splitPrefix_Name.prefix !== 'xs') {
+        var nsName = this.definitions.xmlns[splitPrefix_Name.prefix];
+        if (splitPrefix_Name.name in this.definitions.schemas[nsName].types) {
+            return this.definitions.schemas[nsName].types[splitPrefix_Name.name];
+        } else if (splitPrefix_Name.name in this.definitions.schemas[nsName].complexTypes) {
+            return this.definitions.schemas[nsName].complexTypes[splitPrefix_Name.name];
+        } else
+            return undefined;
+    } else
+        return splitPrefix_Name.prefix + ':' + splitPrefix_Name.name;
+};
+
+WSDL.prototype.findxsiTypeSchema = function (obj) {
+    // Pick up the schema for the type specified in element's xsi:type attribute.
+    var xsiTypeSchema;
+    if (obj[this.options.attributesKey]) {
+        var xsiType = obj[this.options.attributesKey]['xsi:type'];
+        if (xsiType) {
+            var type = splitQName(xsiType);
+            var typeURI;
+            if (type.prefix === TNS_PREFIX) {
+                // In case of xsi:type = "MyType"
+                typeURI = this.xmlns[type.prefix] || this.xmlns.xmlns;
+            } else {
+                typeURI = this.xmlns[type.prefix];
+            }
+            var typeDef = this.findSchemaObject(typeURI, type.name);
+            if (typeDef) {
+                xsiTypeSchema = typeDef.description(this.definitions);
+            }
+        }
+    }
+    return xsiTypeSchema;
+};
+
+WSDL.prototype.findSchema = function (topSchema, name, originalName) {
+    var schema;
+    if (topSchema) {
+        // Find the current object's schema
+        if (topSchema[name + '[]'])
+            name = name + '[]';
+
+        if (topSchema.name) {
+            if (topSchema.name === 'complexType') {
+                schema = this.findSchema(topSchema.children[0], name, originalName);
```

Comment:
As `topSchema.children[0]` is used a few times within this function I think it would make sense to extract it to it's own variable `firstChild`(?), so we don't have to lookup the array everytime for the first child...